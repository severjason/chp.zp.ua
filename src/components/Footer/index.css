.chp-footer-wrapper {
  background-color: #222f3e;
  height: 30rem; }
