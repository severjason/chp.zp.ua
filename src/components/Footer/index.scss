// sass-lint:disable force-element-nesting
// sass-lint:disable no-important
@import '../../scss/variables';

.chp-footer-wrapper {
  background-color: $chp-white-color;
  // height: $chp-container-min-height;

  .ui.grid {
    .chp-footer-element {
      padding: 0;
    }
  }

  .ui.grid.chp-footer-phone {
    background-color: $chp-main-light-color;
    height: $chp-container-min-height / 3;
    margin-top: $chp-container-min-height / 3;

    &::before {
      background-color: $chp-main-light-color;
      bottom: 0;
      content: '';
      height: $chp-container-min-height / 3;
      margin-top: $chp-container-min-height / 3;
      position: absolute;
      right: 0;
      top: 0;
      width: 100vw;
    }

    .chp-icon {
      color: $chp-white-color;
      font-size: 3.5rem;
      height: 3.5rem;
      padding-top: 3rem;
    }

    .chp-text {
      font-size: 1.2rem;
      padding-bottom: 1rem;
    }

    .chp-phone {
      display: block;
      font-size: 2rem;
    }
  }

  .ui.grid.chp-footer-mail {
    background-color: $chp-main-dark-color;
    color: $chp-white-color;
    height: 2 * $chp-container-min-height / 3;

    &::before {
      background-color: $chp-main-dark-color;
      bottom: 0;
      content: '';
      height: 2 * $chp-container-min-height / 3;
      left: 0;
      position: absolute;
      top: 0;
      width: 100vw;
    }

    .chp-icon {
      font-size: 3.5rem;
      height: 3.5rem;
      padding-top: 2rem;
    }

    .chp-mail {
      display: block;
      font-size: 2rem;
      line-height: 3.5rem;
    }

    .chp-copy {
      margin-left: 10%;
    }

    a {
      color: $chp-white-color;
      transition: color .5s;

      &:hover {
        color: $chp-main-light-color;
      }
    }
  }
}

@media all and (max-width: 1200px) {

  .chp-footer-wrapper {
    .ui.grid.chp-footer-mail {
      .chp-copy {
        margin-left: 6.25%;
      }
    }
  }
}

@media all and (max-width: 991px) {

  .chp-footer-wrapper {

    .ui.container.chp-full-width {
      margin: 0 !important;
      width: 100% !important;
    }

    .ui.grid.chp-footer-mail {
      height: $chp-container-min-height / 3;

      &::before {
        width: 0;
      }

      .chp-copy {
        margin: 0 auto;
        text-align: center;
      }
    }

    .ui.grid.chp-footer-phone {
      margin-top: 0;

      &::before {
        width: 0;
      }
    }

  }

}

@media all and (max-width: 600px) {

  .chp-footer-wrapper {

    .ui.container {
      .row > .four.column {
        text-align: center;
      }
    }

    .ui.grid.chp-footer-mail {
      .chp-icon {
        font-size: 3rem;
        height: 3rem;
        padding-top: 1rem;
      }

      .chp-mail {
        font-size: 1.5rem;
        line-height: 3rem;
      }
    }

    .ui.grid.chp-footer-phone {

      .chp-icon {
        font-size: 3rem;
        height: 3rem;
        padding-top: 2.5rem;
      }

      .chp-phone {
        font-size: 1.5rem;
      }
    }
  }
}
