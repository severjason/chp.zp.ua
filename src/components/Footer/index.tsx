import * as React from 'react';
import FooterStyle from './style';

const Footer: React.StatelessComponent = () => {
    return (
        <FooterStyle>
            <div className="chp-footer-wrapper">
                <div className="ui container chp-full-width">
                    <div className="ui grid">
                        <div className="sixteen wide mobile eight wide computer column chp-footer-element">
                            <div className="ui middle aligned centered grid chp-footer-phone">
                                <div className="row">
                                    <div className="right aligned four wide column">
                                        <i className="chp-icon phone icon"/>
                                    </div>
                                    <div className="left aligned twelve wide column">
                                        <div className="chp-text">Позвоните нам:</div>
                                        <a href="tel:+380504864980" className="chp-phone">+38 (050) 486-49-80</a><br/>
                                        <a href="tel:+380637862478" className="chp-phone">+38 (063) 786-24-78</a>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="sixteen wide mobile eight wide computer column chp-footer-element">
                            <div className="ui middle aligned grid chp-footer-mail">
                                <div className="row">
                                    <div className="right aligned four wide column">
                                        <i className="chp-icon envelope outline icon"/>
                                    </div>
                                    <div className="left aligned twelve wide column">
                                        <a href="mailto:yraseverinov@gmail.com" className="chp-mail">
                                            yraseverinov@gmail.com
                                        </a>
                                    </div>
                                </div>
                                <div className="row">
                                    <div className="sixteen wide column chp-copy">
                                        <a href="http://chp.zp.ua/">
                                            chp.zp.ua
                                        </a> @ 2019
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </FooterStyle>
    );
};

export default Footer;
