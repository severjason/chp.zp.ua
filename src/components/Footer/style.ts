import styled from 'styled-components';

const FooterStyle = styled.div`
.chp-footer-wrapper {
  background-color: ${props => props.theme.whiteColor};
  // height: $chp-container-min-height;

  .ui.grid {
    .chp-footer-element {
      padding: 0;
    }
  }

  .ui.grid.chp-footer-phone {
    background-color: ${props => props.theme.mainLightColor};
    height: ${props => props.theme.containerMinHeight / 3}rem;
    margin-top: ${props => props.theme.containerMinHeight / 3}rem;

    &::before {
      background-color: ${props => props.theme.mainLightColor};
      bottom: 0;
      content: '';
      height: ${props => props.theme.containerMinHeight / 3}rem;
      margin-top: ${props => props.theme.containerMinHeight / 3}rem;
      position: absolute;
      right: 0;
      top: 0;
      width: 100vw;
    }

    .chp-icon {
      color: ${props => props.theme.whiteColor};
      font-size: 3.5rem;
      height: 3.5rem;
      padding-top: 3rem;
    }

    .chp-text {
      font-size: 1.2rem;
      padding-bottom: 1rem;
    }

    .chp-phone {
      display: block;
      font-size: 2rem;
    }
  }

  .ui.grid.chp-footer-mail {
    background-color: ${props => props.theme.mainDarkColor};
    color: ${props => props.theme.whiteColor};
    height: ${props => 2 * props.theme.containerMinHeight / 3}rem;

    &::before {
      background-color: ${props => props.theme.mainDarkColor};
      bottom: 0;
      content: '';
      height: ${props => 2 * props.theme.containerMinHeight / 3}rem;
      left: 0;
      position: absolute;
      top: 0;
      width: 100vw;
    }

    .chp-icon {
      font-size: 3.5rem;
      height: 3.5rem;
      padding-top: 2rem;
    }

    .chp-mail {
      display: block;
      font-size: 2rem;
      line-height: 3.5rem;
    }

    .chp-copy {
      margin-left: 10%;
    }

    a {
      color: ${props => props.theme.whiteColor};
      transition: color .5s;

      &:hover {
        color: ${props => props.theme.mainLightColor};
      }
    }
  }
}

@media all and (max-width: 1200px) {

  .chp-footer-wrapper {
    .ui.grid.chp-footer-mail {
      .chp-copy {
        margin-left: 6.25%;
      }
    }
  }
}

@media all and (max-width: 991px) {

  .chp-footer-wrapper {

    .ui.container.chp-full-width {
      margin: 0 !important;
      width: 100% !important;
    }

    .ui.grid.chp-footer-mail {
      height: ${props => props.theme.containerMinHeight / 3}rem;

      &::before {
        width: 0;
      }

      .chp-copy {
        margin: 0 auto;
        text-align: center;
      }
    }

    .ui.grid.chp-footer-phone {
      margin-top: 0;

      &::before {
        width: 0;
      }
    }

  }

}

@media all and (max-width: 600px) {

  .chp-footer-wrapper {

    .ui.container {
      .row > .four.column {
        text-align: center;
      }
    }

    .ui.grid.chp-footer-mail {
      .chp-icon {
        font-size: 3rem;
        height: 3rem;
        padding-top: 1rem;
      }

      .chp-mail {
        font-size: 1.5rem;
        line-height: 3rem;
      }
    }

    .ui.grid.chp-footer-phone {

      .chp-icon {
        font-size: 3rem;
        height: 3rem;
        padding-top: 2.5rem;
      }

      .chp-phone {
        font-size: 1.5rem;
      }
    }
  }
}
`;

export default FooterStyle;