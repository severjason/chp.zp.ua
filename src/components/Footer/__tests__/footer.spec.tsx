import Footer from '../';
import { shallow } from 'enzyme';
import React from 'react';

test('render a footer', () => {
    const wrapper = shallow(<Footer/>);
    expect(wrapper).toMatchSnapshot();
});
