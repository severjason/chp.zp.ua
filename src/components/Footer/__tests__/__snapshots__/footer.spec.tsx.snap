// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a footer 1`] = `
<styled.div>
  <div
    className="chp-footer-wrapper"
  >
    <div
      className="ui container chp-full-width"
    >
      <div
        className="ui grid"
      >
        <div
          className="sixteen wide mobile eight wide computer column chp-footer-element"
        >
          <div
            className="ui middle aligned centered grid chp-footer-phone"
          >
            <div
              className="row"
            >
              <div
                className="right aligned four wide column"
              >
                <i
                  className="chp-icon phone icon"
                />
              </div>
              <div
                className="left aligned twelve wide column"
              >
                <div
                  className="chp-text"
                >
                  Позвоните нам:
                </div>
                <a
                  className="chp-phone"
                  href="tel:+380504864980"
                >
                  +38 (050) 486-49-80
                </a>
                <br />
                <a
                  className="chp-phone"
                  href="tel:+380637862478"
                >
                  +38 (063) 786-24-78
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sixteen wide mobile eight wide computer column chp-footer-element"
        >
          <div
            className="ui middle aligned grid chp-footer-mail"
          >
            <div
              className="row"
            >
              <div
                className="right aligned four wide column"
              >
                <i
                  className="chp-icon envelope outline icon"
                />
              </div>
              <div
                className="left aligned twelve wide column"
              >
                <a
                  className="chp-mail"
                  href="mailto:yraseverinov@gmail.com"
                >
                  yraseverinov@gmail.com
                </a>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="sixteen wide column chp-copy"
              >
                <a
                  href="http://chp.zp.ua/"
                >
                  chp.zp.ua
                </a>
                 @ 
                2019
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;
